### PR TITLE
Add co-authoring commits section to code-review

### PR DESCRIPTION
**Pull request by joshwlambert**
This PR adds a section detailing how to give recognition to collaborators through co-authoring commits. It is similar to 
the [As commit co-authors section](https://epiverse-trace.github.io/blueprints/contribution-acknowledgements.html#as-commit-co-authors) in the Roles in package DESCRIPTION page, but the new additions give more details on some of the technical aspects of how to co-author commits. 

This is information I've compiled that has helped me when suggestions are made in a PR but the changes are made on a different branch.

### CODE REVIEW

File: code-review.qmd [hunk 1]
```diff
@@ -24,3 +24,19 @@ It is left to the maintainer or one of the authors of a package to merge the PR
...
+
+However, in a scenario where the suggested change cannot be commited to the feature branch in the PR, and instead is incorporated in a different feature branch, the suggestion should still be acknowledged. This is where manually specifying [commit coauthors](https://github.blog/2018-01-29-commit-together-with-co-authors/) can help. When using the suggestions of a collaborator, include two empty lines after the commit message and use the phrase `Co-authored-by:` followed by the collaborator's GitHub name and email (this may be a Github no-reply email). See <https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors> for details.
+
+If you do not have the collaborators email address, one way to find it is to navigate on GitHub to a commit they have authored. The URL of the commit should follow the form: `https://github.com/<gh_username>/<repo_name>/commit/<commit_sha>`. Once on this page, add `.patch` to the end of the URL and their email should be given. As stated above, this can be a GitHub no-reply email.
```
Comment (Bisaloo) [inline code-review.qmd:34]: This is good info but I think it'd make more sense to put this in the [Roles in package DESCRIPTION](https://epiverse-trace.github.io/blueprints/contribution-acknowledgements.html) chapter.

File: code-review.qmd [hunk 2]
```diff
@@ -24,3 +24,19 @@ It is left to the maintainer or one of the authors of a package to merge the PR
...
 
 If suggested changes fall outside the scope of the PR, utilise Github's feature of generating issues from PR comments. Clicking the ellipsis in the PR comment and selecting "Reference in new issue" will open an issue with reference to the PR.
+
+## Recognising contributions in reviews
+
+GitHub's feature of suggestions that can be directly committed within the PR are a useful feature that we recommend using for relatively small changes to the code. In most circumstances, these can be directly committed to the feature branch in the PR. GitHub attributes [co-authorship to those that suggested the change and who commited the change](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes). This is a nice feature that enables easy recognition of contributions.
```
Comment (Bisaloo) [inline code-review.qmd:30]:
```suggestion
GitHub's feature of suggestions that can be directly committed within the PR are a useful feature that we recommend using for relatively small changes to the code. In most circumstances, these can be directly committed to the feature branch in the PR. GitHub attributes [co-authorship to those that suggested the change and who commited the change](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes). This is a nice feature that [enables easy recognition of contributions.](contribution-acknowledgements.qmd)
```

File: code-review.qmd [hunk 3]
```diff
@@ -24,3 +24,19 @@ It is left to the maintainer or one of the authors of a package to merge the PR
...
+
+If you do not have the collaborators email address, one way to find it is to navigate on GitHub to a commit they have authored. The URL of the commit should follow the form: `https://github.com/<gh_username>/<repo_name>/commit/<commit_sha>`. Once on this page, add `.patch` to the end of the URL and their email should be given. As stated above, this can be a GitHub no-reply email.
+
+If you have forgotten to include a co-author on a commit that they collaborated on and already commited or pushed, you can [amend your commit message to include them](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/changing-a-commit-message). _Warning: amending old commits can require force pushing, so should be done consideration of the effects_.
```
Comment (Bisaloo) [inline code-review.qmd:36]: We should try and keep things concise and to the point. Generic git tips that are not specific to the topic at hand should be omitted here and possibly added somewhere else.

File: code-review.qmd (hunk 1 above)
Comment (joshwlambert) [reply code-review.qmd:34]: To be precise, it is just L34 you would like to move to Roles in package DESCRIPTION?

File: code-review.qmd (hunk 3 above)
Comment (joshwlambert) [reply code-review.qmd:36]: Removing L36 or moving it to a new section of _Blueprints_? There doesn't look to be any  existing sections that could use this information.

File: code-review.qmd (hunk 1 above)
Comment (Bisaloo) [reply code-review.qmd:34]: Oh right, that was not super clear. Yes, please.

File: code-review.qmd (hunk 3 above)
Comment (Bisaloo) [reply code-review.qmd:36]: It'd probably removing it entirely. It reads more like training material content, which is not really the scope of the blueprints.

Comment (joshwlambert) [reply code-review.qmd]: Resolved in b3a90ef30336ae757a8c4228b927c92ebb72e52a.

**Review by Bisaloo — APPROVED** [review]
Thanks!

**Comment by pratikunterwegs** [comment]
Only seeing this now as I've been away. I'm not sure I understand the use-case for the procedure decribed here. In reviews, accepted reviewer suggestions such as typo fixes are automatically co-authored commits, and any larger contributions are typically as PRs which also credit authors. Would it help to add to this section to say when manual crediting is needed - is there a common case I'm overlooking?

**Comment by joshwlambert** [comment]
Agreed that in most cases, co-authorship is automatic by GitHub, however, when using suggestions in another PR the manual co-authorship is required: the use-case is outlined in Blueprints here: https://github.com/epiverse-trace/blueprints/blob/main/code-review.qmd#L32

**Comment by pratikunterwegs** [comment]
Ah I see thanks, sounds good.

**Pull request merged**